### ISSUE TITLE
Build a per-container qtable for load assessment

### ISSUE THREAD

**Issue by lukemartinlogan** [iowarp/clio-core#318]
@AGENTS.md

I want to make better dynamic scheduling decisions. The goal is to build a model that, given the task's self-defined computation score, reasonably predicts the task's CPU time.

# Containers and Modeling
## DefineModel
Let's build a table of linear models in the static container.
We should add an autogenerated virtual function that defines the initial state of the model called DefineModel. This should be called in the PoolManager when creating the static container. The container should store one model per exposed method. Update chi repo refresh to add this method.
The table will be in the format:
[method_id] [a]

a is just a 32-bit float scalar.

## InferenceModel
Inputs: Task and bool 
The model inference is like this: a * X. Where X is task->stat_.compute_

## ReinforceModel
Inputs: Task, Predicted, Real
Let's use a predefined learning rate of .2.
e = predicted - real
a' <- a - LR * (e / x)

Let's make the LR a configurable parameter, but by default equal to .2. 
The learning_rate should be a new parameter in the chimaera config under runtime key.
Update context-runtime/config/chimaera_default.yaml to be correct.

# Worker Load
Worker should have a float called load that estimates CPU time of currently-executing tasks.
The load is the estimated amount of CPU time of tasks actively executing.

During ExecTask, if the task is just beginning, we will use the model to estimate the load of the task. This will be stored in the RunContext in addition to a CpuTimer, which is a new class we should add to context transport primitives timer.h. CpuTimer calculates only the time spent on CPU. This timer will be called before and after ExecTask to measure the time spent on the coroutine. The time should be accumulating, since tasks may block. This CpuTime should be stored in the RunContext for the task.

During EndTask, we will subtract the load of the task from the system. We will also calculate the true CPU time spent on the task.
